lib/ansiblelint/utils.py

```diff
@@ -732,3 +739,108 @@ def get_rule_skips_from_line(line):
         noqa_text = line.split('# noqa')[1]
         rule_id_list = noqa_text.split()
     return rule_id_list
+
+
+def is_playbook(filename):
+    """
+    Given a filename, it should return true if it looks like a playbook. The
+    function is not supposed to raise exceptions.
+    """
+    # we assume is a playbook if we loaded a sequence of dictionaries where
+    # at least one of these keys is present:
+    playbooks_keys = {
+        "gather_facts",
+        "hosts",
+        "import_playbook",
+        "post_tasks",
+        "pre_tasks",
+        "roles"
+        "tasks",
+    }
+
+    # makes it work with Path objects by converting them to strings
+    if not isinstance(filename, six.string_types):
+        filename = str(filename)
+
+    try:
+        f = parse_yaml_from_file(filename)
+        if isinstance(f, AnsibleSequence) and playbooks_keys.intersection(f[0].keys()):
```

Comment:
This belongs to the else-block

Reply:
@webknjaz Can you please suggest how you see it working? If I move it to a try-else block it will be more verbose and buggy because f[0] could throw a IndexError itself if the sequence is empty. Current code does run ok with it.

Reply:
This the current implementation it's totally unobvious what exceptions each expression can raise. You could do a nested try/except inside. Or even suppress CM. Alternatively, do `next(iter(f), {}})`.